### ISSUE TITLE
Email error and not being sent when referring to expert

### ISSUE THREAD

**Issue by cubitmg** [BiologicalRecordsCentre/iRecord#1571]
![iRecord message 2](https://github.com/BiologicalRecordsCentre/iRecord/assets/65334323/b4808b63-1f89-4e0a-afaa-2373d48d0f93)

**Comment by johnvanbreda** [comment]
@BirenRathod @andrewvanbreda I suspect this is the email problem Andrew discovered on Pantheon, where the PHP mail() function just returns false. Was there a solution?

**Comment by andrewvanbreda** [comment]
Hi @johnvanbreda @BirenRathod 
There wasn't a solution because switching to use Drupal mail didn't support html without a module from what I could work out.
Details here, am hoping Biren will be able to investigate a fix for getting mail() to work on Pantheon systems.
https://github.com/BiologicalRecordsCentre/iRecord/issues/1522

**Comment by BirenRathod** [comment]
@johnvanbreda, @andrewvanbreda just to let you know that if sending email through using the indicia warehouse server then there is a limitation of using that. The server can only 500 emails in one go.

On Pantheon, if the AWS S3 email firewall is not configured using the email address the site has then it refuses to render them.

**Comment by johnvanbreda** [comment]
@BirenRathod it's the Drupal instance that is failing, not the warehouse. Can we check if the AWS S3 firewall is blocking them?

**Comment by BirenRathod** [comment]
Is this plantportal website?

**Comment by johnvanbreda** [comment]
No, iRecord.

**Comment by BirenRathod** [comment]
@johnvanbreda is this still an issue?

**Comment by kitenetter** [comment]
@BirenRathod last week we had another report of iRecord failing to send emails when inviting a user to join an activity. I tested it and have replicated it - as far as I can see no email is being sent. I think this is the same issue as #1522 ?

**Comment by BirenRathod** [comment]
@kitenetter Is this one of the incidents or regularly occurring? I just tested the reset password one and it works fine. If I'm correctly understanding this, the activity one sends through the warehouse's email server.

**Comment by kitenetter** [comment]
@BirenRathod I tested this again by sending an activity invitation to both my home and work emails a couple of hours ago. So far nothing has arrived, but I don't know if there is usually a delay.

**Comment by BirenRathod** [comment]
@kitenetter Can you try again at 13:15, please? or let me know when will be this next possible.

**Comment by kitenetter** [comment]
Tried again at 13:58. No emails have arrived as yet.

**Comment by BirenRathod** [comment]
@kitenetter before trying this can you let me know, so I can track down the error.

**Comment by kitenetter** [comment]
@BirenRathod I will try again at 16:00.

**Comment by kitenetter** [comment]
Just sent

**Comment by kitenetter** [comment]
For info I was masquerading as Drupal user ID 32003 (= warehouse ID 135212) and I sent the invitations using this page:
https://irecord.org.uk/activities/invite?group_id=1773

**Comment by BirenRathod** [comment]
Thanks, @kitenetter. 

@johnvanbreda I found this error -> https://irecord.org.uk/admin/reports/dblog/event/3 There are quite a few similar to this but not sure if it is related to what @kitenetter has experienced.

**Comment by johnvanbreda** [comment]
@BirenRathod I've just fixed that error - there's a new iform module release which will fix some PHP warnings so please can you update? I don't think it relates to the issue with emails being sent though.

**Comment by BirenRathod** [comment]
Thanks @johnvanbreda. I will update the iForm module as soon as possible. 

@kitenetter I'm looking into the cause of that error. I will let you know as soon as I found something.

**Comment by BirenRathod** [comment]
@kitenetter I have done testing using the same link you posted above and I saw the message on the page saying "invitation emails sent" but I haven't received anything.
I also checked errors in the warehouse's server but so far no luck.

**Comment by BirenRathod** [comment]
@johnvanbreda is there a separate smtp config involved in it?

**Comment by johnvanbreda** [comment]
@BirenRathod The send invites function just uses the PHP mail() function. Isn't this the same problem that Andrew reported a while back, where the PHP mail() function no longer works on Pantheon?

**Comment by BirenRathod** [comment]
@johnvanbreda No that was not the same problem. However, that problem had been resolved by using the "Mail system" module.

**Comment by andrewvanbreda** [comment]
Just to confirm the problem I previously reported on Plant Portal with the mail() function has been resolved and that site is operating correctly now.

**Comment by johnvanbreda** [comment]
@andrewvanbreda did you resolve it by changing to use the Mail System module? If so do we need to do that for all Indicia client side code which calls PHP mail()? Or was the resolution something at the Pantheon end?

**Comment by BirenRathod** [comment]
@johnvanbreda The Mail system module works in conjunction with the SMTP module. I helped to configure it on Plantportal and the same has been enabled on iRecord too but still user doesn't receive email.

**Comment by BirenRathod** [comment]
@johnvanbreda Which email address do you use to replay the email? it should be this -> irecord@ceh.ac.uk

**Comment by andrewvanbreda** [comment]
Hi @johnvanbreda
It is something Biren did that fixed it.
We did originally talk about switching to using Drupal mail instead of PHP mail(), however we decided not to because to use HTML mails required a separate module.
The only change I did made was some alterations to group_send_invites to make it compatible with later versions of Drupal. I can't remember the exact changes, but I think they work ok as they seem to be ok on Plant Portal and I think they are in client_helpers Master.
I can't remember exact changes but I think some of the changes involved changing things like
$account = \Drupal::currentUser();

**Comment by johnvanbreda** [comment]
I think it's group_send_invites that isn't working on iRecord in this instance. @BirenRathod the reply from email address is the irecord one you suggest so it's not that. 

I just tried the iRecord verification contact an expert option to email myself and I did receive the email. This uses the same PHP mail() function so it seems that the ability to send an email from Indicia code using mail() does actually work now. However I inspected the email headers and found the following: 
```
X-Spam-Status: No, hits=5.698 tagged_above=-999 required=7
 tests=HDRS_LCASE_IMGONLY=0.099, HTML_IMAGE_ONLY_20=0.7, HTML_MESSAGE=0.001,
 HTML_MIME_NO_HTML_TAG=0.635, HTML_SHORT_LINK_IMG_3=0.328, MIME_HTML_ONLY=0.1,
 MISSING_DATE=1.396, MISSING_MID=0.14, RDNS_NONE=1,
 TO_NO_BRKTS_NORDNS_HTML=1.298, T_REMOTE_IMAGE=0.01,
 T_SCC_BODY_TEXT_LINE=-0.01, URIBL_BLOCKED=0.001
```

My x-Spam limit is set to 7 which is quite high - many people won't receive an email with a score over 5 so it is highly likely that similar emails will get blocked as spam.  There are a few things we should do in the code according to this:
1. Add a `<html>` tag to the body.
2. Add a text only version in a separate mime part (if possible).
3. Add the date to the header.
4. Add a Message ID to the header

I think RDNS_NONE is at server level so I can't fix that. I will try and review the uses of the mail() function with this in mind to see what I can do.
